packages/shared/blockchain/btc-based/src/lib/btc-based.wallet.utils.ts

```diff
@@ -3,18 +3,23 @@ import { generateMnemonic, mnemonicToSeed } from 'bip39'
 import { ECPair, payments } from 'bitcoinjs-lib'
 import hdkey from 'hdkey'
 import { TronWallet } from '@tatumio/api-client'
-import { BtcBasedBlockchain, blockchainUtils } from '@tatumio/shared-core'
+import { BtcBasedBlockchain, blockchainUtils, Blockchain } from '@tatumio/shared-core'
 
 export const btcBasedWalletUtils = {
   generateAddressFromXPub: (
     blockchain: BtcBasedBlockchain,
     xpub: string,
     i: number,
     options?: { testnet: boolean },
+    addressType: 'bech32' | 'p2sh' = 'bech32',
   ): string => {
     const network = blockchainUtils.getNetworkConfig(blockchain, options)
     const w = fromBase58(xpub, network).derivePath(String(i))
-    return payments.p2pkh({ pubkey: w.publicKey, network }).address as string
+    if (addressType === 'bech32' && blockchain === Blockchain.BTC) {
```

Comment:
Wouldn't it be better to have custom implementation of BTC? Not for all BTC chains.

Reply:
The addressType param is gone now, I'd say the if(BTC) makes sense for a btc-based utility (?), it is implemented in a very similar way in core-api for change address generation (btc currency check).